core/src/test/scala/org/apache/spark/HeartbeatReceiverSuite.scala

```diff
@@ -287,6 +287,8 @@ private class FakeSchedulerBackend(
     resourceProfileManager: ResourceProfileManager)
   extends CoarseGrainedSchedulerBackend(scheduler, rpcEnv) {
 
+  def this() = this(null, null, null, null)
```

Comment:
I still have no idea how this wasn't required in Scala 2.12, as it's used with a no-arg constructor but none existed ?!